feed/script.js

```diff
@@ -0,0 +1,205 @@
+      content = formatOOORequests(data);
+      break;
+    default:
+      content = 'Unknown activity type';
+  }
+  item.innerHTML = `
+    ${content}`;
+  return item;
+}
+
+function formatOOORequests(data) {
+  let text = '';
+  if (data.type === logType.REQUEST_CREATED)
+    text = `<p><strong>${data?.user || data?.createdBy
+      }</strong> raised an OOO request from <strong>${new Date(
+        data.from,
+      ).toLocaleString()}</strong> until <strong>${new Date(
+        data.until,
+      ).toLocaleString()}</strong></p>`;
+  if (data.status === 'APPROVED')
+    text = `${data?.user || data?.createdBy
+      } approved an extension request for task:`;
+  if (data.status === 'DENIED')
+    text = `${data?.user || data?.createdBy
+      } rejected an extension request for task:`;
+
+  return `<div>
+            <div class="title">
+              <img src="assets/leave.png" class="img_icon">
+              <p class="">${text}</p>
+           </div>
+           ${data.message ? `<strong>Reason:</strong> ${data.message}` : ''}
+           <p class="timestamp">
+          ${new Date(data.timestamp * 1000)}
+          </p>
+       </div>`;
+}
+
+function formatExtensionRequestsLog(data) {
+  let text = '';
+  if (data.status === 'PENDING')
+    text = `${data?.user || data?.createdBy
+      } raised an extension request for task:`;
+  if (data.status === 'APPROVED')
+    text = `${data?.user || data?.createdBy
+      } approved an extension request for task:`;
+  if (data.status === 'DENIED')
+    text = `${data?.user || data?.createdBy
+      } rejected an extension request for task:`;
+  return `<div>
+            <div class="title">
+              <img src="assets/extensionReq.png" class="img_icon">
+              <p class="">${text}</p>
+           </div>
+           <p>
+           Task:- 
+          <a href=${STATUS_BASE_URL}/tasks/${data?.taskId}>${data.title || 'Untitled Task'
+    }</a>
+          </p>
+          <p class="timestamp">
+          ${new Date(data.timestamp * 1000)}
+          </p>
+       </div>`;
+}
+
+function formatTasksLog(data) {
+  let text = '';
+  if (data.status)
+    text = `${data?.user || data?.createdBy
+      } changed the status of the task to <strong>${data?.status}</strong>`;
+  if (data.percentCompleted)
+    text = text
+      ? `${text} and changed the task progress to <strong>${data?.percentCompleted}%`
+      : `${data?.user || data?.createdBy
+      } updated the task progress to <strong>${data?.percentCompleted}%`;
+
+  return `<div>
+            <div class="title">
+              <img src="assets/task.png" class="img_icon">
+              <p class="">${text}</p>
+            </div>
+            <p> Task:- 
+                <a href=${STATUS_BASE_URL}/tasks/${data?.taskId}>${data.title || 'Untitled Task'
+    }</a>
+           </p>
+            <p class="timestamp">${new Date(data.timestamp * 1000)}</p>
+       </div>`;
+}
+
+async function populateActivityFeed(query = {}, newLink) {
```

Comment:
The check if (activityFeedContainer.innerHTML === '') within the finally block might not be entirely reliable. It only checks if the innerHTML is empty, but it could potentially contain whitespace or other non-breaking elements. You could refine the check to ensure there's actual data to be displayed before adding the empty page message.

Reply:
Done